src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -135,6 +135,20 @@ export class IndexPatternsService {
     return this.savedObjectsCache.map((obj) => obj?.attributes?.title);
   };
 
+  find = async (search: string, size: number = 10): Promise<IndexPattern[]> => {
+    const savedObjects = await this.savedObjectsClient.find<IndexPatternSavedObjectAttrs>({
+      type: 'index-pattern',
+      fields: ['title'],
+      search: `${search}*`,
```

Comment:
This should use the plain `search` value and the caller of the method can append the `*` if desired.